srml/im-online/src/lib.rs

```diff
@@ -488,10 +546,6 @@ impl<T: Trait> session::OneSessionHandler<T::AccountId> for Module<T> {
 		};
 
 		T::ReportUnresponsiveness::report_offence(vec![], offence);
-
-		// Remove all received heartbeats from the current session, they have
-		// already been processed and won't be needed anymore.
-		<ReceivedHeartbeats>::remove_prefix(&<session::Module<T>>::current_index());
```

Comment:
This was a bug, we would not cleanup the session heartbeats if there was no offence to report.